src/main/java/Main.java

```diff
@@ -1,16 +1,30 @@
-import java.util.Scanner;
+import java.io.IOException;
+import java.util.*;
 
 public class Main {
 
-    // nie zmieniaj nic w main
     public static void main(String[] args) {
         Scanner scanner = new Scanner(System.in);
         Main main = new Main();
         main.run(scanner);
     }
 
     void run(Scanner scanner) {
-        // usupełnij metodę
+        CountryReader countryReader = new CountryReader();
+        try {
+            HashMap<String, Country> countryHashMap = countryReader.readCountries("countries.csv");
+            System.out.println("Podaj kod kraju, o którym chcesz zobaczyć informacje:");
+            String countryName = scanner.nextLine().toUpperCase();
+            if (countryHashMap.containsKey(countryName)) {
+                Country country = countryHashMap.get(countryName);
+                System.out.println(country);
+            } else {
+                System.out.println("Kod kraju " + countryName + " nie został znaleziony.");
+            }
+        } catch (IOException e) {
+            System.out.println("Brak pliku countries.csv.");
+        }
+        scanner.close();
```

Comment:
Nie zamykaj scannera opartego o `System.in`, Scanner może działać również na plikach, wtedy zamykanie ma sens